### DISCUSSION TITLE
Tailwind & Alpine: Menu Items Won't Stay Visible When Clicked

### DISCUSSION THREAD

**Discussion by sbarats-nrg** [tailwindlabs/tailwindcss#10733] [Help]
I'm very new to Alpinejs and only have some experience with Tailwind. Can anyone explain why my navbar submenus aren't functioning properly? I would also like the first submenu to be hidden on page load. Thank you for any help!

Tailwind Play link: https://play.tailwindcss.com/n4oYjc0inE

It doesn't seem that I can add JS to the link, so this is what I have:

```
import Alpine from "alpinejs";

Alpine.data('nav', () => ({
    menuOpened: false,
    menuInvestOpened: false,
    menuLearnOpened: false,
    menuInsightsOpened: false,
    menuCompanyOpened: false,

    mobileMenuOpened: false,
    mobileInvestOpened: false,
    mobileLearnOpened: false,
    mobileInsightsOpened: false,
    mobileCompanyOpened: false,

    toggle(investMenu) {
        this.menuInvestOpened = investMenu;
        
    }
}))
```

**Comment by wongjn** [comment]
You seem to be missing the `Alpine.start()` call to [initialize Alpine when used in bundled code](https://alpinejs.dev/essentials/installation#as-a-module). You seem to be missing an element with the `x-data="nav"` to correspond with your `Alpine.data('nav', …)` registration.

---

> I would also like the first submenu to be hidden on page load

You could try using the [cloak directive](https://alpinejs.dev/directives/cloak):

> Sometimes, when you're using AlpineJS for a part of your template, there is a "blip" where you might see your uninitialized template after the page loads, but before Alpine loads.
>
> `x-cloak` addresses this scenario by hiding the element it's attached to until Alpine is fully loaded on the page.

---

Consider seeking assistance in a discussion area more geared towards Alpine.js, since it seems to be that aspect that you seem to be struggling with here.

**Reply by sbarats-nrg** [reply]
Seems I forgot to add it to the `<body>` in my Play code - sorry about that.

Adding `Alpine.start()` caused the menu items to lose its colors and the icons disappear, so does this mean it's already being bundled?

Adding `x-cloak` did not hide the submenu, unfortunately.

I will follow your suggestion and look into this on more Alpine-focused forums, but I greatly appreciate your help.
